deploy/flux-config-map.yaml

```diff
@@ -0,0 +1,15 @@
+---
```

Comment:
If this gets applied again, it will overwrite a value that's been updated by something else (e.g., a sidecar, or a controller, creating a docker config for ECR). Perhaps I'm misunderstanding how you're intending this to be used, though?

Reply:
Hey good catch. So the problem with ECR atm is that the token is running out after 12hrs. So I would need to recreate it. But maybe i can use a variable here to make it a bit more flexible. Or do you have a better idea of how to make the config.json file available but overwriteable from outside?

Reply:
A btw @squaremo. Do i need to make those charts adjustment in deploy, charts or both? Was confusing me a lot that there actually 2 places

Reply:
The hard requirement is that the secret exists at the time pods are created, so it can be mounted. I think it's OK if another process goes and updates the secret _after_ it's been created.

If the secret doesn't contain the actual file, that is also OK. I will need to make sure that `fluxd` doesn't break if the docker-config file isn't present; but otherwise, if you update the secret from another pod or by hand, the file will be created or updated in the volume (after some minutes), and `fluxd` will be able to use it.

Crucially, if the secret doesn't contain a `data` field, it won't overwrite the existing contents when you reapply the manifest (e.g., if you upgrade the chart). We rely on this for the SSH key generation (see `deploy/flux-secret.yaml`).

Reply:
> Do i need to make those charts adjustment in deploy, charts or both?

You can just add it to the chart if you wish. The files in `deploy/` are just examples, for people who aren't using helm -- if you do add things there, consider comments explaining what the various bits are.

Reply:
Will fix this one here. Actually the example should also use a secret